java/core/src/java/org/apache/orc/TypeDescription.java

```diff
@@ -565,9 +557,7 @@ public int getScale() {
    * @return a list of sorted attribute names
    */
   public List<String> getAttributeNames() {
-    List<String> result = new ArrayList<>(attributes.keySet());
-    Collections.sort(result);
-    return result;
+    return new ArrayList<>(attributes.keySet());
```

Comment:
Keep the `attributes` Map sorted so it doesn't have to be sorted every time this method is invoked.